### PR TITLE
Add Lock View related documents

### PR DESCRIPTION
**Pull request by MyonKeminta**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

* Added document about the feature Lock View
  * The three new tables in infomation_schema
  * The configuration
  * The change to the lock conflict troubleshooting document.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [x] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: information-schema/information-schema.md [hunk 1]
```diff
@@ -55,16 +55,20 @@ Information Schema 提供了一种查看系统元数据的 ANSI 标准方法。
...
 | [`CLUSTER_HARDWARE`](/information-schema/information-schema-cluster-info.md)            | 提供在每个 TiDB 组件上发现的底层物理硬件的详细信息。           |
 | [`CLUSTER_INFO`](/information-schema/information-schema-cluster-info.md)                | 提供当前集群拓扑的详细信息。                                   |
 | [`CLUSTER_LOAD`](/information-schema/information-schema-cluster-load.md)                | 提供集群中 TiDB 服务器的当前负载信息。                         |
 | [`CLUSTER_LOG`](/information-schema/information-schema-cluster-log.md)                  | 提供整个 TiDB 集群的日志。                                     |
 | `CLUSTER_PROCESSLIST`                                                                   | 提供 `PROCESSLIST` 表的集群级别的视图。                        |
 | `CLUSTER_SLOW_QUERY`                                                                    | 提供 `SLOW_QUERY` 表的集群级别的视图。                         |
 | `CLUSTER_STATEMENTS_SUMMARY`                                                            | 提供 `STATEMENTS_SUMMARY` 表的集群级别的视图。                 |
-| `CLUSTER_STATEMENTS_SUMMARY_HISTORY`                                                    | 提供 `CLUSTER_STATEMENTS_SUMMARY_HISTORY` 表的集群级别的视图。 |
```
Comment (MyonKeminta) [inline information-schema/information-schema.md:65]: I think this is a mistake so I changed it 🤔

**Review by longfangsong — APPROVED** [review]

**Comment by TomShawn** [comment]
@youjiali1995 PTAL. Thanks!

File: information-schema/information-schema-data-lock-waits.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+2 rows in set (0.01 sec)                                                                
+```
+
+上述结果显示，事务 ID 为 `425405024158875649` 的事务在执行 Digest 为 `"f7530877a35ae65300c42250abd8bc731bbaf0a7cabc05dab843565230611bb22"` 的语句的过程中，试图在 `"7480000000000000355f728000000000000002"` 这个 key 上获取悲观锁，但是该 key 目前被事务 ID 为 `425405016242126849` 的事务持有。
+
+## SQL Digest
+
+`DATA_LOCK_WAITS` 表中会记录 SQL Digest，并不记录 SQL 原文。
+
+SQL Digest 是 SQL 归一化之后的哈希值。对于最近一段时间内执行过的语句，可以从 `STATEMENTS_SUMMARY` 或 `STATEMENTS_SUMMARY_HISTORY` 中根据 Digest 查找到对应的归一化 SQL 的原文：
```
Comment (youjiali1995) [inline information-schema/information-schema-data-lock-waits.md:67]: 需不需要提一下 cluster 表？

File: information-schema/information-schema-deadlocks.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
++--------------------+---------------------+------+------+---------+-------+ 
+```
+
+`DEADLOCKS` 表中需要用多行来表示同一个死锁事件，每行显示参与死锁的其中一个事务的信息。当该 TiDB 节点记录了多次死锁错误时，需要按照 `DEADLOCK_ID` 列来区分，相同的 `DEADLOCK_ID` 表示同一个死锁事件。需要注意，`DEADLOCK_ID` **并不保证全局唯一，也不会持久化**，因而其只能在同一个结果集里表示同一个死锁事件。
+
+`DEADLOCKS` 表的各列的含义如下：
+
+* `DEADLOCK_ID`：死锁事件的 ID。当表内存在多次死锁错误的信息时，需要使用该列来区分属于不同死锁错误的行。
+* `OCCUR_TIME`：发生该次死锁错误的时间。
+* `RETRYABLE`：该次死锁错误是否可重试。目前暂不支持收集可充式的死锁错误的信息，因而该字段恒为 0。在一些特定的情况下，一个死锁错误可能是可重试的，此时 TiDB 内部会自动重试。
```
Comment (youjiali1995) [inline information-schema/information-schema-deadlocks.md:41]:
```suggestion
* `RETRYABLE`：该次死锁错误是否可重试。目前暂不支持收集可重试的死锁错误的信息，因而该字段恒为 0。在一些特定的情况下，一个死锁错误可能是可重试的，此时 TiDB 内部会自动重试。
```
特定情况可能要介绍一下，然后说明下记录这个信息的原因是发生可重试死锁不会报错但会影响性能。

File: information-schema/information-schema-data-lock-waits.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+| KEY                    | varchar(64)         | NO   |      | NULL    |       |
+| TRX_ID                 | bigint(21) unsigned | NO   |      | NULL    |       |
+| CURRENT_HOLDING_TRX_ID | bigint(21) unsigned | NO   |      | NULL    |       |
+| SQL_DIGEST             | varchar(64)         | YES  |      | NULL    |       |
++------------------------+---------------------+------+------+---------+-------+
+```
+
+`DATA_LOCK_WAITS` 表的各列的含义如下：
+
+* `KEY`：正在发生等锁的 KEY，以十六进制编码的形式显示。
```
Comment (youjiali1995) [inline information-schema/information-schema-data-lock-waits.md:34]: tidb 现在有办法能解析这种 key 是哪个表哪一行或者索引吗？没有的话下个 sprint 要优化一下体验，可能用户知道了 sql 也没用，还要知道是哪一行才行。

**Review by youjiali1995 — COMMENTED** [review]
需不需要介绍一下这几张表能做什么、解决什么问题之类的？有些问题还需要这几张表一起用才能解决。

File: information-schema/information-schema-data-lock-waits.md (hunk 4 above)
Comment (MyonKeminta) [reply information-schema/information-schema-data-lock-waits.md:34]: 这个后续迭代可以考虑再加个字段展示把 key 解出来的信息。目前可能得借助些其它的工具才能解。我等会看下 tidb-ctl 顶用不顶用，能用的话晚点我把它补充进文档里。

**Comment by MyonKeminta** [comment]
I'll continue adding contents tomorrow

File: information-schema/information-schema-data-lock-waits.md (hunk 4 above)
Comment (MyonKeminta) [reply information-schema/information-schema-data-lock-waits.md:34]: tidb-ctl 好像不行...

File: troubleshoot-lock-conflicts.md [hunk 5]
```diff
@@ -238,4 +238,148 @@ TiDB 在使用悲观锁的情况下，多个事务之间出现了死锁，必定
...
+| key                                    | count |
++----------------------------------------+-------+
+| 7480000000000000415f728000000000000001 |     2 |
+| 7480000000000000415f728000000000000002 |     1 |
++----------------------------------------+-------+
+```
+
+可以考虑多次查询以避免偶然性。
+
+如果已知频繁出问题的 key，便可尝试从 `TIDB_TRX` / `CLUSTER_TIDB_TRX` 表中获取试图在该 key 上上锁的事务的信息。需要注意 `TIDB_TRX` 和 `CLUSTER_TIDB_TRX` 表所展示的信息也是对其进行查询的时刻正在运行的事务的信息，并不展示已经结束的事务。如果并发的事务数量很大，该查询的结果集也可能很大，可以考虑添加 limit 子句，或用 where 子句筛选出等锁时间较长的事务。需要注意，对 Lock View 中的多张表进行 join 时，不同表之前的数据并不保证在同一时刻获取，因而可能不同表中的信息可能并不同步。
```
Comment (youjiali1995) [inline troubleshoot-lock-conflicts.md:317]:
```suggestion
如果已知频繁出问题的 key，便可尝试从 `TIDB_TRX` / `CLUSTER_TIDB_TRX` 表中获取试图在该 key 上上锁的事务的信息。需要注意 `TIDB_TRX` 和 `CLUSTER_TIDB_TRX` 表所展示的信息也是对其进行查询的时刻正在运行的事务的信息，并不展示已经结束的事务。如果并发的事务数量很大，该查询的结果集也可能很大，可以考虑添加 limit 子句，或用 where 子句筛选出等锁时间较长的事务。需要注意，对 Lock View 中的多张表进行 join 时，不同表之间的数据并不保证在同一时刻获取，因而不同表中的信息可能并不同步。

```

File: troubleshoot-lock-conflicts.md [hunk 6]
```diff
@@ -238,4 +238,148 @@ TiDB 在使用悲观锁的情况下，多个事务之间出现了死锁，必定
...
+              USER: root
+                DB: test
+   ALL_SQL_DIGESTS: [0fdc781f19da1c6078c9de7eadef8a307889c001e05f107847bee4cfc8f3cdf3, a4e28cc182bdd18288e2a34180499b9404cd0
+ba07e3cc34b6b3be7b7c2de7fe9]
+2 rows in set (0.00 sec)
+```
+
+#### 事务被长时间阻塞
+
+如果已知一个事务被另一事务（或多个事务）阻塞，且已知当前事务的 start ts（即事务 ID），则可使用如下方式获取导致该事务阻塞的事务的信息。注意对 Lock View 中的多张表进行 join 时，不同表之前的数据并不保证在同一时刻获取，因而可能不同表中的信息可能并不同步。
```
Comment (youjiali1995) [inline troubleshoot-lock-conflicts.md:359]:
```suggestion
如果已知一个事务被另一事务（或多个事务）阻塞，且已知当前事务的 start ts（即事务 ID），则可使用如下方式获取导致该事务阻塞的事务的信息。注意对 Lock View 中的多张表进行 join 时，不同表之间的数据并不保证在同一时刻获取，因而可能不同表中的信息可能并不同步。

```

File: troubleshoot-lock-conflicts.md [hunk 7]
```diff
@@ -238,4 +238,148 @@ TiDB 在使用悲观锁的情况下，多个事务之间出现了死锁，必定
...
+
+{{< copyable "sql" >}}
+
+```sql
+select trx.* from information_schema.data_lock_waits as l left join information_schema.tidb_trx as trx on l.trx_id = trx.id where l.key = "7480000000000000415f728000000000000001"\G
+```
+
+```
+mysql> select trx.* from information_schema.data_lock_waits as l left join information_schema.tidb_trx as trx on l.trx_id =
+trx.id where l.key = "7480000000000000475f728000000000000001"\G
```
Comment (youjiali1995) [inline troubleshoot-lock-conflicts.md:327]:
```suggestion
```

**Review by youjiali1995 — APPROVED** [review]

**Comment by MyonKeminta** [comment]
@TomShawn PTAL

**Comment by TomShawn** [comment]
@MyonKeminta All newly added documents need an entry in https://github.com/pingcap/docs-cn/blob/master/TOC.md. Otherwise, we cannot find them in the left navigation menu.

File: information-schema/information-schema-deadlocks.md [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+假设有如下表定义和初始数据：
+
+{{< copyable "sql" >}}
+
+```sql
+create table t (id int primary key, v int);
+insert into t values (1, 10), (2, 20);
+```
+
+令两个事务按如下顺序执行：
```
Comment (TomShawn) [inline information-schema/information-schema-deadlocks.md:60]:
```suggestion
使两个事务按如下顺序执行：
```

File: information-schema/information-schema-deadlocks.md [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+|           1 | 2021-06-04 08:22:38.765699 |         0 | 425405959304904707 | 22230766411edb40f27a68dadefc63c6c6970d5827f1e5e22fc97be2c4d8350d | 7480000000000000385F728000000000000002 | 425405959304904708 |
+|           1 | 2021-06-04 08:22:38.765699 |         0 | 425405959304904708 | 22230766411edb40f27a68dadefc63c6c6970d5827f1e5e22fc97be2c4d8350d | 7480000000000000385F728000000000000001 | 425405959304904707 |
++-------------+----------------------------+-----------+--------------------+------------------------------------------------------------------+----------------------------------------+--------------------+
+```
+
+该表中产生了两行数据，两行的 `DEADLOCK_ID` 字段皆为 1，表示这两行数据包含同一次死锁错误的信息。第一行显示事务 ID 为 `425405959304904707` 的事务，在 `"7480000000000000385F728000000000000002"` 这个 key 上，被事务 ID 为 `"425405959304904708"` 的事务阻塞了；第二行则显示事务 ID 为 `"425405959304904708"` 的事务在 `"7480000000000000385F728000000000000001"` 这个 key 上被事务 ID 为 `425405959304904707` 的事务阻塞了，构成了相互阻塞的状态，形成了死锁。
+
+## 示例 2
+
+假设查询 `DEADLOCKS` 表获得了如下结果集：
```
Comment (TomShawn) [inline information-schema/information-schema-deadlocks.md:90]:
```suggestion
假设查询 `DEADLOCKS` 表得到了如下结果集：
```

File: information-schema/information-schema-deadlocks.md [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
++-------------+----------------------------+-----------+--------------------+------------------------------------------------------------------+----------------------------------------+--------------------+
+```
+
+上述结果中的 `DEADLOCK_ID` 表明，前两行共同表示一次死锁错误的信息，两条事务相互等待构成了死锁；而后三行共同表示另一次死锁信息，三个事务循环等待构成了死锁。
+
+## 可重试的死锁错误
+
+> **注意：**
+>
+> 当前版本 `DEADLOCKS` 表暂不支持收集可重试的死锁错误相关的信息。
```
Comment (TomShawn) [inline information-schema/information-schema-deadlocks.md:110]: 当前版本是指 v5.1 吗

File: information-schema/information-schema-deadlocks.md [hunk 11]
```diff
@@ -0,0 +1,188 @@
...
+
+当一个事务 A 被另一个事务已经持有的锁阻塞，而另一个事务 B 直接或间接地被当前事务持有的锁阻塞，将会引发一个死锁错误。这里，事务 B 可能（直接或间接地）被事务 A 之前已经执行完的语句产生的锁阻塞，也可能被事务 A 目前正在执行的语句阻塞。对于前者，TiDB 将会向事务 A 的客户端报告死锁错误，并终止该事务；而对于后者，事务 A 当前正在执行的语句将在 TiDB 内部被自动重试。例如，假设事务 A 执行了如下语句：
```
Comment (TomShawn) [inline information-schema/information-schema-deadlocks.md:112]:
```suggestion
当事务 A 被另一个事务已经持有的锁阻塞，而事务 B 直接或间接地被当前事务持有的锁阻塞，将会引发一个死锁错误。这里：

+ 事务 B 可能（直接或间接地）被事务 A 之前已经执行完的语句产生的锁阻塞
+ 事务 B 也可能被事务 A 目前正在执行的语句阻塞

对于前者，TiDB 将会向事务 A 的客户端报告死锁错误，并终止该事务；而对于后者，事务 A 当前正在执行的语句将在 TiDB 内部被自动重试。例如，假设事务 A 执行了如下语句：
```

File: information-schema/information-schema-deadlocks.md [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+3. 事务 B 执行第二条语句试图对 `id = 1` 的行上锁，被事务 A 阻塞
+4. 事务 A 试图对 `id = 2` 的行上锁，被 B 阻塞，形成死锁
+
+对于这种情况，由于事务 A 阻塞其它事务的语句也是当前正在执行的语句，因而可以解除当前语句所上的悲观锁（使得事务 B 可以继续运行），并重试当前语句。TiDB 内部使用 key 的 hash 来判断是否属于这种情况。
+
+当可重试的死锁发生时，内部自动重试并不会引起事务报错，因而对客户端透明，但是这种情况的频繁发生可能影响性能。当这种情况发生时，在 TiDB 的日志中可以观察到 `single statement deadlock, retry statement` 字样的日志。
+
+## CLUSTER_DEADLOCKS
+
+`CLUSTER_DEADLOCKS` 表是 `DEADLOCKS` 的集群版本，其返回整个集群上的每个 TiDB 节点中最近发生数次的死锁错误（每个节点上的 `DEADLOCKS` 表内的信息合并到一起）。`CLUSTER_DEADLOCKS` 包含额外的 `INSTANCE` 列展示所属节点的 IP 地址和端口，用以区分不同的 TiDB 节点。
```
Comment (TomShawn) [inline information-schema/information-schema-deadlocks.md:142]:
```suggestion
`CLUSTER_DEADLOCKS` 表是 `DEADLOCKS` 的集群版本，返回整个集群上每个 TiDB 节点中最近发生数次的死锁错误（将每个节点上的 `DEADLOCKS` 表内的信息合并到一起）。`CLUSTER_DEADLOCKS` 包含额外的 `INSTANCE` 列展示所属节点的 IP 地址和端口，用以区分不同的 TiDB 节点。
```

**Review by TomShawn — COMMENTED** [review]
Good writing! These are some minor comments.

File: information-schema/information-schema-tidb-trx.md [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+| DB                 | varchar(64)                                             | YES  |      | NULL    |       |
+| ALL_SQL_DIGESTS    | text                                                    | YES  |      | NULL    |       |
++--------------------+---------------------------------------------------------+------+------+---------+-------+
+```
+
+## SQL Digest
+
+`TIDB_TRX` 表中会记录 SQL Digest，并不记录 SQL 原文。
+
+SQL Digest 是 SQL 归一化之后的哈希值。对于当前 TiDB 最近一段时间内执行过的语句，可以从 `STATEMENTS_SUMMARY` 或 `STATEMENTS_SUMMARY_HISTORY` 中根据 Digest 查找到对应的归一化 SQL 的原文。注意 `STATEMENTS_SUMMARY` 和 `STATEMENTS_SUMMARY_HISTORY` 只在当前 TiDB 节点范围内查询；如需在整个集群范围内查询，则可使用 `CLUSTER_STATEMENTS_SUMMARY` 和 `CLUSTER_STATEMENTS_SUMMARY_HISTORY` 代替。
```
Comment (qiancai) [inline information-schema/information-schema-tidb-trx.md:119]:
```suggestion
SQL Digest 是 SQL 归一化之后的哈希值。如需查找 SQL Digest 对应的 SQL 原文，请进行以下操作之一：

- 对于当前 TiDB 节点在最近一段时间内执行过的语句，你可以从 `STATEMENTS_SUMMARY` 或 `STATEMENTS_SUMMARY_HISTORY` 中根据 SQL Digest 查找到对应的 SQL 原文。
- 对于整个集群所有 TiDB 节点在最近一段时间内执行过的语句，你可以从 `CLUSTER_STATEMENTS_SUMMARY` 或`CLUSTER_STATEMENTS_SUMMARY_HISTORY` 中根据 SQL Digest 查找到对应的 SQL 原文。
```

File: information-schema/information-schema-tidb-trx.md [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+
+```
++------------------------------------------------------------------+---------------------------------------+
+| digest                                                           | digest_text                           |
++------------------------------------------------------------------+---------------------------------------+
+| f7530877a35ae65300c42250abd8bc731bbaf0a7cabc05dab843565230611bb2 | update `t` set `v` = ? where `id` = ? |
++------------------------------------------------------------------+---------------------------------------+
+```
+
+关于 SQL Digest 和 `STATEMENTS_SUMMARY`、`STATEMENTS_SUMMARY_HISTORY` 表的详细说明请参阅 [Statement Summary Tables](/statement-summary-tables.md)。
```
Comment (qiancai) [inline information-schema/information-schema-tidb-trx.md:135]:
```suggestion
关于 SQL Digest 和 `STATEMENTS_SUMMARY`、`STATEMENTS_SUMMARY_HISTORY` 、`CLUSTER_STATEMENTS_SUMMARY`、`CLUSTER_STATEMENTS_SUMMARY_HISTORY` 表的详细说明，请参阅 [Statement Summary Tables](/statement-summary-tables.md)。
```

File: information-schema/information-schema-deadlocks.md (hunk 10 above)
Comment (MyonKeminta) [reply information-schema/information-schema-deadlocks.md:110]: 是的

Comment (MyonKeminta) [reply information-schema/information-schema-deadlocks.md:110]: 这里用“目前”这个说法怎样？毕竟也不知道这个行为将来在哪个版本会改

File: information-schema/information-schema-tidb-trx.md [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+   MEM_BUFFER_KEYS: 2
+  MEM_BUFFER_BYTES: 48
+        SESSION_ID: 7
+              USER: root
+                DB: test
+   ALL_SQL_DIGESTS: [e6f07d43b5c21db0fbb9a31feac2dc599787763393dd5acbfad80e247eb02ad5, 04fa858fa491c62d194faec2ab427261cc7998b3f1ccf8f6844febca504cb5e9, f7530877a35ae65300c42250abd8bc731bbaf0a7cabc05dab843565230611bb2]
+1 row in set (0.00 sec)
+```
+
+上述查询结果返回当前节点有一个事务正在执行。其信息表示该事务处于闲置状态（`CURRENT_SQL_DIGEST` 为 `NULL`），`ALL_SQL_DIGESTS` 显示 了其曾经执行过的 3 条语句的 Digest。
```
[the remark this answers is not in the document]
Comment (MyonKeminta) [reply information-schema/information-schema-tidb-trx.md:83]: 该事务已经执行过 3 条语句，ALL_SQL_DIGESTS 字段显示了这三条语句的 digest

**Comment by MyonKeminta** [comment]
PTAL again thanks!

File: information-schema/information-schema-deadlocks.md [hunk 16]
```diff
@@ -0,0 +1,196 @@
...
+```
+
+`DEADLOCKS` 表中需要用多行来表示同一个死锁事件，每行显示参与死锁的其中一个事务的信息。当该 TiDB 节点记录了多次死锁错误时，需要按照 `DEADLOCK_ID` 列来区分，相同的 `DEADLOCK_ID` 表示同一个死锁事件。需要注意，`DEADLOCK_ID` **并不保证全局唯一，也不会持久化**，因而其只能在同一个结果集里表示同一个死锁事件。
+
+`DEADLOCKS` 表中各列的字段含义如下：
+
+* `DEADLOCK_ID`：死锁事件的 ID。当表内存在多次死锁错误的信息时，需要使用该列来区分属于不同死锁错误的行。
+* `OCCUR_TIME`：发生该次死锁错误的时间。
+* `RETRYABLE`：该次死锁错误是否可重试。目前暂不支持收集可重试的死锁错误的信息，因而该字段值恒为 0。关于可重试的死锁错误的说明，参见[可重试的死锁错误](#可重试的死锁错误)小节。
+* `TRY_LOCK_TRX_ID`：试图上锁的事务 ID，即事务的 `start_ts`。
```
[the remark this answers is not in the document]
Comment (MyonKeminta) [reply information-schema/information-schema-deadlocks.md:42]: 搜了一下目前已有的文档里的写法就不统一，两种写法都出现过，带下划线的居多。我这个 PR 里面的内容应该已经全都写成 \`start_ts\` 了。按我个人理解的话，用反引号括起来的话表示代码，需要带下划线，如果不带下划线的话，反引号也可以去掉

File: information-schema/information-schema-deadlocks.md [hunk 17]
```diff
@@ -0,0 +1,196 @@
...
+```
+
+那么如果事务 A 先后对 `id = 1` 和 `id = 2` 的两行分别上锁，且两个事务以如下时序运行：
+
+1. 事务 A 对 `id = 1` 的行上锁
+2. 事务 B 执行第一条语句并对 `id = 2` 的行上锁
+3. 事务 B 执行第二条语句试图对 `id = 1` 的行上锁，被事务 A 阻塞
+4. 事务 A 试图对 `id = 2` 的行上锁，被 B 阻塞，形成死锁
+
+对于这种情况，由于事务 A 阻塞其它事务的语句也是当前正在执行的语句，因而可以解除当前语句所上的悲观锁（使得事务 B 可以继续运行），并重试当前语句。TiDB 内部使用 key 的 hash 来判断是否属于这种情况。
```
[the remark this answers is not in the document]
Comment (MyonKeminta) [reply information-schema/information-schema-deadlocks.md:141]: 是的

File: tidb-configuration-file.md [hunk 18]
```diff
@@ -599,6 +599,13 @@ TiDB 服务状态相关配置。
 + 悲观事务中单个语句最大重试次数，重试次数超过该限制，语句执行将会报错。
 + 默认值：256
 
+### deadlock-history-capacity
+
++ 单个 TiDB 节点的 [`INFORMATION_SCHEMA.DEADLOCKS`](/information-schema/information-schema-deadlocks.md) 表最多可记录的死锁事件个数。当表的容量已满时，如果再次发生死锁错误，最旧的一次死锁错误的信息将从表中移除。
```
Comment (qiancai) [inline tidb-configuration-file.md:604]:
```suggestion
+ 单个 TiDB 节点的 [`INFORMATION_SCHEMA.DEADLOCKS`](/information-schema/information-schema-deadlocks.md) 表最多可记录的死锁事件个数。当表的容量已满时，如果再次发生死锁错误，最早的一次死锁错误的信息将从表中移除。
```

File: information-schema/information-schema-deadlocks.md [hunk 19]
```diff
@@ -0,0 +1,196 @@
...
```
Comment (TomShawn) [inline information-schema/information-schema-deadlocks.md:141]:
```suggestion
对于情况二，由于事务 A 阻塞其它事务的语句也是当前正在执行的语句，因而可以解除当前语句所上的悲观锁（使得事务 B 可以继续运行），并重试当前语句。TiDB 内部使用 key 的 hash 来判断是否属于这种情况。
```

**Review by TomShawn — APPROVED** [review]
LGTM

File: information-schema/information-schema-data-lock-waits.md (hunk 4 above)
Comment (TomShawn) [reply information-schema/information-schema-data-lock-waits.md:34]: 见 https://github.com/pingcap/docs/pull/5798#discussion_r653234850

File: information-schema/information-schema-data-lock-waits.md [hunk 20]
```diff
@@ -0,0 +1,86 @@
...
+```sql
+*************************** 1. row ***************************                          
+                   KEY: 7480000000000000355f728000000000000002                          
+                TRX_ID: 425405024158875649                                              
+CURRENT_HOLDING_TRX_ID: 425405016242126849                                              
+            SQL_DIGEST: f7530877a35ae65300c42250abd8bc731bbaf0a7cabc05dab843565230611bb22
+2 rows in set (0.01 sec)                                                                
+```
+
+以上查询结果显示，ID 为 `425405024158875649` 的事务在执行 Digest 为 `"f7530877a35ae65300c42250abd8bc731bbaf0a7cabc05dab843565230611bb22"` 的语句的过程中，试图在 `"7480000000000000355f728000000000000002"` 这个 key 上获取悲观锁，但是该 key 目前被 ID 为 `425405016242126849` 的事务持有。
```
Comment (TomShawn) [inline information-schema/information-schema-data-lock-waits.md:61]: 见 https://github.com/pingcap/docs/pull/5798#discussion_r653237329

File: information-schema/information-schema-deadlocks.md [hunk 21]
```diff
@@ -0,0 +1,196 @@
...
+
+以上查询结果中的 `DEADLOCK_ID` 列表明，前两行共同表示一次死锁错误的信息，两条事务相互等待构成了死锁；而后三行共同表示另一次死锁信息，三个事务循环等待构成了死锁。
+
+## 可重试的死锁错误
+
+> **注意：**
+>
+> 目前，`DEADLOCKS` 表暂不支持收集可重试的死锁错误相关的信息。
+
+当事务 A 被另一个事务已经持有的锁阻塞，而事务 B 直接或间接地被当前事务持有的锁阻塞，将会引发一个死锁错误。这里：
```
Comment (TomShawn) [inline information-schema/information-schema-deadlocks.md:112]: 见 https://github.com/pingcap/docs/pull/5798#discussion_r653255145

**Comment by TomShawn** [comment]
见 https://github.com/pingcap/docs/pull/5798#discussion_r655922056

**Comment by TomShawn** [comment]
见 https://github.com/pingcap/docs/pull/5798#discussion_r655831929

**Comment by TomShawn** [comment]
All followup comments are addressed in https://github.com/pingcap/docs-cn/pull/6577/files.

**Pull request merged**